packages/react/src/date-picker/Calendar.tsx

```diff
@@ -192,6 +189,57 @@ const calendarComponents: CustomComponents = {
 			</Box>
 		);
 	},
+	// Custom `Row` component to improve accessibility of markup from react-day-picker
+	// Default: https://github.com/gpbl/react-day-picker/blob/main/src/components/Row/Row.tsx
+	Row: function Row(props: RowProps) {
+		const { styles, classNames, components } = useDayPicker();
+		const DayComponent = components?.Day;
+		return (
+			<tr className={classNames.row} style={styles.row}>
+				{props.dates.map((date) => (
+					<td
+						key={getUnixTime(date)}
+						className={classNames.cell}
+						style={styles.cell}
+						role="gridcell"
+					>
+						{DayComponent ? (
+							<DayComponent displayMonth={props.displayMonth} date={date} />
+						) : null}
+					</td>
+				))}
+			</tr>
+		);
+	},
+	// Custom `Day` component to improve accessibility of markup from react-day-picker
+	// Default: https://github.com/gpbl/react-day-picker/blob/main/src/components/Day/Day.tsx
+	Day: function Day(props: DayProps) {
+		const buttonRef = useRef<HTMLButtonElement>(null);
+		const dayRender = useDayRender(props.date, props.displayMonth, buttonRef);
+
+		if (dayRender.isHidden) {
+			return null;
+		}
+
+		if (!dayRender.isButton) {
+			return <div {...dayRender.divProps} role={undefined} />;
```

Comment:
Not sure why this `role` is here?
```suggestion
			return <div {...dayRender.divProps} />;
```

Reply:
Oh, does `divProps` include a `gridcell` `role`?

Reply:
@stowball yeah it does :(

Reply:
I've added a comment to make this clearer